specs/schemes/exact/scheme_exact_algo.md

```diff
@@ -233,23 +120,154 @@ Algorand uses [**msgpack**](https://www.npmjs.com/package/algorand-msgpack) to e
+- Public keys, with a sha512_256 checksum (last 4 bytes) appended as a suffix. This encoding ensures that addresses can be derived directly from public keys without requiring a pre-existing signature for operations like `ecRecover` to verify signatures.
+- SHA512_256 of the Logic Signatures program bytecode + `Program` as a prefix.
+
+Example of encoding Algorand addresses:
+
+```ts
+	encodeAddress(publicKey: Buffer): string {
+		const keyHash: string = sha512_256.create().update(publicKey).hex()
+
+		// last 4 bytes of the hash
+		const checksum: string = keyHash.slice(-8)
+
+		return base32.encode(Encoder.ConcatArrays(publicKey, Buffer.from(checksum, "hex"))).slice(0, 58)
+	}
+```
+
+### Encoding
+
+Algorand uses [**msgpack**](https://www.npmjs.com/package/algorand-msgpack) to encode the transactions. Each element in the `paymentGroup` array can be base64-decoded, then msgpack-decoded, to reveal the transaction contents. Or using `goal` you can do the following:
+
+```json
+% cat payload.json | jq -r '.paymentGroup[]' | base64 -d | goal clerk inspect -
+-[0]
+{
+  "txn": {
+    "fee": 2000,
+    "fv": 53347179,
+    "gen": "mainnet-v1.0",
+    "gh": "wGHE2Pwdvd7S12BL5FaOP20EGYesN73ktiC1qzkkit8=",
+    "grp": "fy1Szr+lgvgTJsviMY2KnHSsXqyfCJ1UOCE+2Tf3vS8=",
+    "lv": 53348179,
+    "rcv": "FACILITATORADDRESSAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAALQCXBZE",
+    "snd": "FACILITATORADDRESSAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAALQCXBZE",
+    "type": "pay"
+  }
+}
+
+-[1]
+{
+  "sig": "/cnUMjoYtJ8rSctl+29LJUwm4U4TjzE+Wdmk10QlZsZtzGwPloBLDNrhONvCGAQXR1c6/wTdb9/ba+QTN8XLBQ==",
+  "txn": {
+    "aamt": 5000000,
+    "arcv": "RESOURCESERVERADDRESSAAAAAAAAAAAAAAAAAAAAAAAAAAAAAALTSRPAE",
+    "fv": 53347179,
+    "gh": "wGHE2Pwdvd7S12BL5FaOP20EGYesN73ktiC1qzkkit8=",
+    "grp": "fy1Szr+lgvgTJsviMY2KnHSsXqyfCJ1UOCE+2Tf3vS8=",
+    "lv": 53348179,
+    "snd": "CLIENTAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAHFUPIRI",
+    "type": "axfer",
+    "xaid": 31566704
+  }
+}
+```
+
+## Appendix
+
+### Gasless Transactions/Sponsored Fees
+
+`Facilitator`s who offer "gasless" transactions, may refuse to sign their `feePayer` transaction, if they consider it's been constructed maliciously. These malicious checks may be off-loaded to a logic signature, where the `facilitator` instead provides their signature of the TxID as an argument.
+
+Example Facilitator Logic Signature:
+
+```TEAL
+#pragma version 12
+
+// Validate feePayer's signature
+txn TxID
+int 0
+args
+addr FACILITATORADDRESSAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAALQCXBZE
+ed25519verify_bare
+assert
+
+// Disallow rekeying
+txn RekeyTo
+global ZeroAddress
+==
+assert
+
+// Disallow closing
+txn CloseRemainderTo
+global ZeroAddress
+==
+assert
+
+// Must be a payment transaction
+txn TypeEnum
+int pay
+==
+assert
+
+// Amount must be 0 Algo
+txn Amount
+int 0
+==
+assert
+
+// Fee must be 2x MinTxnFee
+txn Fee
+global MinTxnFee
+int 2
+*
+==
+assert
+
+// Approve
+int 1
+```
```

Comment:
I think in the example we should mandate a stricter condition on the "allowed" fee abstraction:

- Must be for a `group` of _exactly_ 2 transactions;
- The "sponsored" transaction must be:
    - An `axfer`
    - Of a specific `asset`
    - With an amount `> 0`
    - With a receiver different from the sender